commit 4a45304cd58ae387cacae554f3c762204f5e9c35
Author: John Doe <jdoe@email.com>
Date:   Sun Feb 11 23:46:45 2024 +0800

    Remove Win 8 fallback handling in DWriteFontProxy    2022-11-18 04:49:46    b2e60f07f8f0b85c9fd803243aefb52fc5dad93d

diff --git a/content/child/dwrite_font_proxy/dwrite_font_proxy_init_impl_win.cc b/content/child/dwrite_font_proxy/dwrite_font_proxy_init_impl_win.cc
index 54d6353caa532..ed3037dffebcf 100644
--- a/content/child/dwrite_font_proxy/dwrite_font_proxy_init_impl_win.cc
+++ b/content/child/dwrite_font_proxy/dwrite_font_proxy_init_impl_win.cc
@@ -79,8 +79,17 @@ void InitializeDWriteFontProxy() {
 
   skia::OverrideDefaultSkFontMgr(std::move(skia_font_manager));
 
-  DCHECK(g_font_fallback);
-  blink::WebFontRendering::SetUseSkiaFontFallback(true);
+  // When IDWriteFontFallback is not available (prior to Win8.1) Skia will
+  // still attempt to use DirectWrite to determine fallback fonts (in
+  // SkFontMgr_DirectWrite::onMatchFamilyStyleCharacter), which will likely
+  // result in trying to load the system font collection. To avoid that and
+  // instead fall back on WebKit's fallback logic, we don't use Skia's font
+  // fallback if IDWriteFontFallback is not available.
+  // This flag can be removed when Win8.0 and earlier are no longer supported.
+  bool fallback_available = g_font_fallback != nullptr;
+  DCHECK_EQ(fallback_available,
+            base::win::GetVersion() > base::win::Version::WIN8);
+  blink::WebFontRendering::SetUseSkiaFontFallback(fallback_available);
 }
 
 void UninitializeDWriteFontProxy() {
